### PR TITLE
DRILL-6929: Exclude maprfs jar for default profile

### PR DESCRIPTION
**Pull request by vvysotskyi**
- Moved dependency on `drill-format-mapr` into mapr profile in `contrib/storage-hive/core/pom.xml`
- Moved class `ConvertHiveMapRDBJsonScanToDrillMapRDBJsonScan` which uses `drill-format-mapr` classes into `scrMapr/main/java` folder
- Specified `scrMapr/main/java` as source directory only for mapr profile using `build-helper-maven-plugin`

For problem description please see [DRILL-6929](https://issues.apache.org/jira/browse/DRILL-6929).

### CODE REVIEW

**Comment by arina-ielchiieva** [comment]
@vdiravka could you please review?

File: contrib/storage-hive/core/pom.xml [hunk 1]
```diff
@@ -142,11 +142,6 @@
         </exclusion>
       </exclusions>
     </dependency>
-    <dependency>
```
Comment (vdiravka) [inline contrib/storage-hive/core/pom.xml]: To follow excluding MapR specific code from the default profile, please move `format-maprdb` to `mapr` profile in the contrib/pom.xml

```
  <modules>
    <module>storage-hbase</module>
    <module>storage-hive</module>
    <module>storage-mongo</module>
    <module>storage-jdbc</module>
    <module>storage-kafka</module>
    <module>storage-kudu</module>
    <module>storage-opentsdb</module>
    <module>data</module>
    <module>udfs</module>
  </modules>
```

```
    <profile>
      <id>mapr</id>
      <modules>
        <module>format-maprdb</module>
      </modules>
    </profile>
```

File: contrib/storage-hive/core/scrMapr/main/java/org/apache/drill/exec/planner/sql/logical/ConvertHiveMapRDBJsonScanToDrillMapRDBJsonScan.java [hunk 2]
```diff
@@ -110,15 +100,15 @@ Hive does not always give correct costing (i.e. for external tables Hive does no
         To ensure Drill MapR-DB Json scan will be chosen, reduce Hive scan importance to 0.
        */
       call.getPlanner().setImportance(hiveScanRel, 0.0);
-    } catch (final Exception e) {
+    } catch (RuntimeException e) {
```
[the remark this answers is not in the document]
Comment (vdiravka) [reply contrib/storage-hive/core/scrMapr/main/java/org/apache/drill/exec/planner/sql/logical/ConvertHiveMapRDBJsonScanToDrillMapRDBJsonScan.java]: Looks like for future we might consider to add trowing `IOException` to all `getFormatPlugin()` methods. And `Throwables.throwIfInstanceOf(e.getCause(), IOException.class);` for `MapRDBFormatPlugin.getFormatPlugin()`. But for now only `MapRDBFormatPlugin` requires it. 
Could you write a short TODO about it?

**Comment by vdiravka** [comment]
+1

File: contrib/storage-hive/core/scrMapr/main/java/org/apache/drill/exec/planner/sql/logical/ConvertHiveMapRDBJsonScanToDrillMapRDBJsonScan.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (vvysotskyi) [reply contrib/storage-hive/core/scrMapr/main/java/org/apache/drill/exec/planner/sql/logical/ConvertHiveMapRDBJsonScanToDrillMapRDBJsonScan.java]: Done.

**Pull request closed without merging**